### 1.  XYZ Company Dataset

This company weekly spends on different marketing channels for a period from 2015 to 2019 is to analyze the trends and patterns in the marketing spend data to identify the most effective channels and optimize the marketing budget allocation.

In this project I have aimed to analyse the data using different parameters and present some data
visualization using some python libraries like numpy, pandas, matplotlib and seaborn.

#### 1.1 Install required libraries

In this section, I’ll be importing the required libraries such as Numpy * Pandas * Matplotlib * Seaborn

In [1]:
import numpy as np
import pandas as pd

# for handling pdf file first install tabula-py packege !pip install tabula-py

import tabula

#### 1.2 Loading the dataset

In [2]:
#Convert PDF Table Into a DataFrame
#declare the path of your file
file_path = "Assignment Dataset - Brandintelle - dataset.pdf"

#Convert pdf file
df = tabula.read_pdf(file_path, pages='all')

In [3]:
df[0]

Sr No        DATE         Sales  TV_Spends  OOH_Spends  Print_Spends  \
0       1  2015-11-23  2.754372e+06   167687.6           0   95463.66667   
1       2  2015-11-30  2.584277e+06   214600.9           0       0.00000   
2       3  2015-12-07  2.547387e+06        0.0      248022    3404.00000   
3       4  2015-12-14  2.875220e+06   625877.3           0  132600.00000   
4       5  2015-12-21  2.215953e+06        0.0      520005       0.00000   
5       6  2015-12-28  2.569922e+06   249189.8           0  239417.33330   
6       7  2016-01-04  2.171507e+06     5152.9      425163       0.00000   
7       8  2016-01-11  2.464132e+06    22141.7           0  176859.66670   
8       9  2016-01-18  2.012520e+06        0.0           0       0.00000   
9      10  2016-01-25  1.738912e+06        0.0           0   84329.33333   
10     11  2016-02-01  1.772307e+06   123838.7       79645       0.00000   
11     12  2016-02-08  1.809058e+06        0.0           0   35726.66667   
12     13  2016-02-15  1.952740e+06   497073.3           0   18363.00000   
13     14  2016-02-22  1.507805e+06        0.0           0       0.00000   
14     15  2016-02-29  1.510392e+06   311672.2           0       0.00000   
15     16  2016-03-07  1.588840e+06        0.0           0  123795.66670   
16     17  2016-03-14  1.605990e+06   148293.3       82489       0.00000   
17     18  2016-03-21  1.356010e+06        0.0           0       0.00000   
18     19  2016-03-28  2.103937e+06        0.0      123916  101845.66670   
19     20  2016-04-04  1.120835e+06        0.0           0       0.00000   
20     21  2016-04-11  1.141140e+06        0.0      124526       0.00000   
21     22  2016-04-18  1.166880e+06    83450.5           0       0.00000   
22     23  2016-04-25  8.888067e+05        0.0           0       0.00000   
23     24  2016-05-02  8.988733e+05        0.0           0   93889.33333   
24     25  2016-05-09  8.893733e+05        0.0       93671       0.00000   
25     26  2016-05-16  1.915375e+06        0.0           0       0.00000   
26     27  2016-05-23  1.006845e+06        0.0      233330       0.00000   
27     28  2016-05-30  8.691983e+05        0.0           0       0.00000   
28     29  2016-06-06  7.272500e+05        0.0           0       0.00000   
29     30  2016-06-13  7.537300e+05        0.0           0       0.00000   
30     31  2016-06-20  9.991967e+05   153542.9      232328       0.00000   
31     32  2016-06-27  8.458433e+05        0.0           0       0.00000   
32     33  2016-07-04  1.126017e+06    75907.6           0   38763.00000   
33     34  2016-07-11  1.140978e+06        0.0           0       0.00000   
34     35  2016-07-18  1.242657e+06    14253.1      599331   35855.00000   
35     36  2016-07-25  1.293483e+06        0.0           0       0.00000   
36     37  2016-08-01  1.484843e+06   211351.9      307907       0.00000   

    FB_Impressions  Paid_Search_Clicks  Search_Spends  competitor_sales_B  \
0     7.290385e+07            0.000000              0             8125009   
1     1.658110e+07        29511.715460          31000             7901549   
2     4.995477e+07        36132.358960          28400             8300197   
3     3.164930e+07        36804.210960          31900             8122883   
4     8.802269e+06        28401.744070          27100             7105985   
5     4.990208e+07        38061.775910          28300             7097237   
6     4.114770e+07        36289.513780          27800             6974368   
7     5.655469e+07        34765.108990          32000             7452174   
8     3.573056e+07        28023.481500          22900             6703566   
9     3.186126e+07        21069.739250          22900             5535394   
10    0.000000e+00        28023.977300          20600             5480098   
11    3.424966e+07        21323.331970          20200             5872523   
12    1.518976e+07            0.000000              0             5822926   
13    0.000000e+00            0.00000

In [4]:
# convert PDF into CSV file
tabula.convert_into("Assignment Dataset - Brandintelle - dataset.pdf", "Assignment_Dataset_Brandintelle.csv", output_format="csv", pages='all')

In [5]:
# Load the dataset
df2 = pd.read_csv("Assignment_Dataset_Brandintelle.csv")

In [6]:
df2

Sr No        DATE        Sales  TV_Spends  OOH_Spends  Print_Spends  \
0        1  2015-11-23  2754371.667   167687.6           0   95463.66667   
1        2  2015-11-30  2584276.667   214600.9           0       0.00000   
2        3  2015-12-07  2547386.667        0.0      248022    3404.00000   
3        4  2015-12-14  2875220.000   625877.3           0  132600.00000   
4        5  2015-12-21  2215953.333        0.0      520005       0.00000   
..     ...         ...          ...        ...         ...           ...   
203    204  2019-10-14  2456240.000        0.0       60433  153723.66670   
204    205  2019-10-21  2182825.000   154917.6           0       0.00000   
205    206  2019-10-28  2377706.667    21982.5       14094   17476.00000   
206    207  2019-11-04  2732825.000    22453.0           0   24051.33333   
207    208  2019-11-11  2767788.333        0.0           0       0.00000   

     FB_Impressions  Paid_Search_Clicks  Search_Spends  competitor_sales_B  \
0      7.290385e+07             0.00000              0             8125009   
1      1.658110e+07         29511.71546          31000             7901549   
2      4.995477e+07         36132.35896          28400             8300197   
3      3.164930e+07         36804.21096          31900             8122883   
4      8.802269e+06         28401.74407          27100             7105985   
..              ...                 ...            ...                 ...   
203    0.000000e+00        152840.32340         112100             7612762   
204    5.688196e+07        103680.04780         103700             6701667   
205    0.000000e+00        138387.70410         114700             7688920   
206    0.000000e+00        151707.99050         134100             8815710   
207    1.279811e+08             0.00000              0             9067759   

        FB_Spends  events  
0    228213.98740      na  
1     34258.57351      na  
2    127691.26130  event0  
3     84014.72031      na  
4     20687.47816      na  
..            ...     ...  
203       0.00000      na  
204  133624.57550      na  
205       0.00000      na  
206       0.00000      na  
207  366190.75940      na  

[208 rows x 12 columns]

### Summary of given data set

This dat|a set about a company they spend different marketing chennels for a period from 2015 to 2019.
This chennels like TV, OOH(Out-of-home advertising), Print, Facebook and Paid Scearch Clicks. In this data set also provide weekly sales through all this channels, and they also provide cometitor sales

identify the most effective channels and optimize the marketing budget allocation.

In this project I have aimed to analyse the data using different parameters and present some data visualization using some python libraries like numpy, pandas, matplotlib and seaborn

The data set contains information on the weekly spends of the XYZ company on different marketing channels from 2015 to 2019. The data is organized by week and includes the amount spent on each marketing channel. The marketing channels may include but are not limited to television, radio, print, online advertising, social media, events, and promotions.

To summarize the data, we can calculate the average weekly spend on each marketing channel over the 5-year period. Additionally, we can analyze the trends and patterns in the spending over time. For example, we can identify which marketing channels had the highest and lowest spending, which channels showed an increase or decrease in spending over time, and whether there were any significant changes in spending during certain periods.

Overall, the data can provide valuable insights into the marketing strategies of the XYZ company and help inform future decisions on where to allocate marketing budgets.

### Data set column descriptions

- DATE: date in YYYY-MM-DD format
- Sales: Weekly revenue figure made by the comapany
- TV_Spends: Weekly spends on TV advertisements
- OOH_Spends: Weekly spends on Outdoor Medium
- Print_Spends: Weekly spends on Print Medium like TV and magzines
- FB_Impressions: Weekly Impressions of ads on Facebook |  impressions can be described as views that   a particular ad received on facebook
- FB_Spends: Weekly spends on Facebook Ads
- Search_Spends: Weekly spends on paid search results. These are spends done to get the paid         
  advertisements on the search results.
- Paid_Search_Clicks : Weekly clicks on the paid advertisement 
- events: Events can be described as festivals or promotional events done by the company. This is a  
  categorical variable
- competitor_sales_B: Sales figure of the competition product


#### Data Preparation & Cleaning


In these preparation section, we will :
- Look at some things like shape information description missing/null values
  duplicated values
- Look the number of rows & columns, ranges of values etc.
- Handle missing, incorrect and invalid data
  This will allow us to understand the data more precisely and make it easier for us to further     analyze it.

In [7]:
df2.head()

Sr No        DATE        Sales  TV_Spends  OOH_Spends  Print_Spends  \
0      1  2015-11-23  2754371.667   167687.6           0   95463.66667   
1      2  2015-11-30  2584276.667   214600.9           0       0.00000   
2      3  2015-12-07  2547386.667        0.0      248022    3404.00000   
3      4  2015-12-14  2875220.000   625877.3           0  132600.00000   
4      5  2015-12-21  2215953.333        0.0      520005       0.00000   

   FB_Impressions  Paid_Search_Clicks  Search_Spends  competitor_sales_B  \
0    7.290385e+07             0.00000              0             8125009   
1    1.658110e+07         29511.71546          31000             7901549   
2    4.995477e+07         36132.35896          28400             8300197   
3    3.164930e+07         36804.21096          31900             8122883   
4    8.802269e+06         28401.74407          27100             7105985   

      FB_Spends  events  
0  228213.98740      na  
1   34258.57351      na  
2  127691.26130  event0  
3   84014.72031      na  
4   20687.47816      na

In [8]:
df2.tail()

Sr No        DATE        Sales  TV_Spends  OOH_Spends  Print_Spends  \
203    204  2019-10-14  2456240.000        0.0       60433  153723.66670   
204    205  2019-10-21  2182825.000   154917.6           0       0.00000   
205    206  2019-10-28  2377706.667    21982.5       14094   17476.00000   
206    207  2019-11-04  2732825.000    22453.0           0   24051.33333   
207    208  2019-11-11  2767788.333        0.0           0       0.00000   

     FB_Impressions  Paid_Search_Clicks  Search_Spends  competitor_sales_B  \
203    0.000000e+00         152840.3234         112100             7612762   
204    5.688196e+07         103680.0478         103700             6701667   
205    0.000000e+00         138387.7041         114700             7688920   
206    0.000000e+00         151707.9905         134100             8815710   
207    1.279811e+08              0.0000              0             9067759   

       FB_Spends events  
203       0.0000     na  
204  133624.5755     na  
205       0.0000     na  
206       0.0000     na  
207  366190.7594     na

In [9]:
df2.sample(10)

Sr No        DATE        Sales  TV_Spends  OOH_Spends   Print_Spends  \
3        4  2015-12-14  2875220.000   625877.3           0  132600.000000   
92      93  2017-08-28  2336431.667   884168.4           0       0.000000   
16      17  2016-03-14  1605990.000   148293.3       82489       0.000000   
46      47  2016-10-10  2406035.000        0.0           0   80428.333330   
182    183  2019-05-20   746205.000   219302.9       10817   28421.666670   
93      94  2017-09-04  2002065.000        0.0           0       0.000000   
200    201  2019-09-23  2510951.667   208627.1           0       0.000000   
203    204  2019-10-14  2456240.000        0.0       60433  153723.666700   
157    158  2018-11-26  2653766.667   401439.0           0  113471.000000   
169    170  2019-02-18  1675816.667        0.0           0    3762.666667   

     FB_Impressions  Paid_Search_Clicks  Search_Spends  competitor_sales_B  \
3       31649297.03         36804.21096          31900             8122883   
92      14227048.49         70335.07472          52500             7010039   
16             0.00         17065.23476          17800             4938630   
46      46050657.17         37640.48678          29900             7711902   
182            0.00             0.00000              0             2251073   
93             0.00         44809.82457          48700             6679241   
200            0.00        145612.54130         115700             7702931   
203            0.00        152840.32340         112100             7612762   
157     65751650.87        156564.38460         120000             7536793   
169            0.00             0.00000              0             5588194   

        FB_Spends events  
3     84014.72031     na  
92    26967.80957     na  
16        0.00000     na  
46   108425.00090     na  
182       0.00000     na  
93        0.00000     na  
200       0.00000     na  
203       0.00000     na  
157  210107.85320     na  
169       0.00000     na

In [10]:
#shape
print('The shape of the dataset:', df2.shape)

The shape of the dataset: (208, 12)


In [11]:
#Information
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 208 entries, 0 to 207
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Sr No               208 non-null    int64  
 1   DATE                208 non-null    object 
 2   Sales               208 non-null    float64
 3   TV_Spends           208 non-null    float64
 4   OOH_Spends          208 non-null    int64  
 5   Print_Spends        208 non-null    float64
 6   FB_Impressions      208 non-null    float64
 7   Paid_Search_Clicks  208 non-null    float64
 8   Search_Spends       208 non-null    int64  
 9   competitor_sales_B  208 non-null    int64  
 10  FB_Spends           208 non-null    float64
 11  events              208 non-null    object 
dtypes: float64(6), int64(4), object(2)
memory usage: 19.6+ KB


In [12]:
df2.isnull()

Sr No   DATE  Sales  TV_Spends  OOH_Spends  Print_Spends  FB_Impressions  \
0    False  False  False      False       False         False           False   
1    False  False  False      False       False         False           False   
2    False  False  False      False       False         False           False   
3    False  False  False      False       False         False           False   
4    False  False  False      False       False         False           False   
..     ...    ...    ...        ...         ...           ...             ...   
203  False  False  False      False       False         False           False   
204  False  False  False      False       False         False           False   
205  False  False  False      False       False         False           False   
206  False  False  False      False       False         False           False   
207  False  False  False      False       False         False           False   

     Paid_Search_Clicks  Search_Spends  competitor_sales_B  FB_Spends  events  
0                 False          False               False      False   False  
1                 False          False               False      False   False  
2                 False          False               False      False   False  
3                 False          False               False      False   False  
4                 False          False               False      False   False  
..                  ...            ...                 ...        ...     ...  
203               False          False               False      False   False  
204               False          False               False      False   False  
205               False          False               False      False   False  
206               False          False               False      False   False  
207               False          False               False      False   False  

[208 rows x 12 columns]

In [13]:
#duplicated values
df2.duplicated().sum()

0

In [14]:
#duplicated values Column 'Sr no' 
df2['Sr No'].duplicated().sum()

0

In [15]:
#decription
df2.describe().T

count          mean           std        min  \
Sr No               208.0  1.045000e+02  6.018859e+01        1.0   
Sales               208.0  1.822143e+06  7.162286e+05   672250.0   
TV_Spends           208.0  1.113277e+05  2.141877e+05        0.0   
OOH_Spends          208.0  8.103364e+04  1.574839e+05        0.0   
Print_Spends        208.0  2.796474e+04  4.862303e+04        0.0   
FB_Impressions      208.0  2.446024e+07  3.509738e+07        0.0   
Paid_Search_Clicks  208.0  5.083562e+04  4.084228e+04        0.0   
Search_Spends       208.0  4.436635e+04  3.526877e+04        0.0   
competitor_sales_B  208.0  5.538025e+06  2.077192e+06  2240235.0   
FB_Spends           208.0  6.436973e+04  9.481091e+04        0.0   

                             25%           50%           75%           max  
Sr No               5.275000e+01  1.045000e+02  1.562500e+02  2.080000e+02  
Sales               1.165211e+06  1.874514e+06  2.378407e+06  3.827520e+06  
TV_Spends           0.000000e+00  0.000000e+00  1.380503e+05  1.185349e+06  
OOH_Spends          0.000000e+00  0.000000e+00  9.535900e+04  9.381780e+05  
Print_Spends        0.000000e+00  0.000000e+00  3.575875e+04  2.394173e+05  
FB_Impressions      0.000000e+00  0.000000e+00  4.121226e+07  1.782983e+08  
Paid_Search_Clicks  1.884205e+04  4.279576e+04  7.571053e+04  1.565644e+05  
Search_Spends       1.765000e+04  3.605000e+04  6.402500e+04  1.341000e+05  
competitor_sales_B  3.589581e+06  5.538524e+06  7.311814e+06  9.984742e+06  
FB_Spends           0.000000e+00  0.000000e+00  1.086904e+05  4.620117e+05

In [16]:
# Columns
df2.columns

Index(['Sr No', 'DATE', 'Sales', 'TV_Spends', 'OOH_Spends', 'Print_Spends',
       'FB_Impressions', 'Paid_Search_Clicks', 'Search_Spends',
       'competitor_sales_B', 'FB_Spends', 'events'],
      dtype='object')

In [17]:
#data types check
df2.dtypes

Sr No                   int64
DATE                   object
Sales                 float64
TV_Spends             float64
OOH_Spends              int64
Print_Spends          float64
FB_Impressions        float64
Paid_Search_Clicks    float64
Search_Spends           int64
competitor_sales_B      int64
FB_Spends             float64
events                 object
dtype: object

Observation
- The dataset contains 208 rows and 12 columns
- Most the columns of the dataset are type float64 dtype
- No duplicates value in this dataset
- No missing values in this dataset
- Some features contains missing values such as director, cast, country, date_added, rating & duration.


In [18]:
# Copy of this dataset 
df2_copy = df2.copy()

In [19]:
df2_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 208 entries, 0 to 207
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Sr No               208 non-null    int64  
 1   DATE                208 non-null    object 
 2   Sales               208 non-null    float64
 3   TV_Spends           208 non-null    float64
 4   OOH_Spends          208 non-null    int64  
 5   Print_Spends        208 non-null    float64
 6   FB_Impressions      208 non-null    float64
 7   Paid_Search_Clicks  208 non-null    float64
 8   Search_Spends       208 non-null    int64  
 9   competitor_sales_B  208 non-null    int64  
 10  FB_Spends           208 non-null    float64
 11  events              208 non-null    object 
dtypes: float64(6), int64(4), object(2)
memory usage: 19.6+ KB


In [20]:
# Change some column data type float type to int type
df2_copy['Sales'] = df2_copy['Sales'].astype('int')
df2_copy['TV_Spends'] = df2_copy['TV_Spends'].astype('int')
df2_copy['Print_Spends'] = df2_copy['Print_Spends'].astype('int')
df2_copy['FB_Impressions'] = df2_copy['FB_Impressions'].astype('int')
df2_copy['Paid_Search_Clicks'] = df2_copy['Paid_Search_Clicks'].astype('int')
df2_copy['FB_Spends'] = df2_copy['FB_Spends'].astype('int')

In [21]:
df2_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 208 entries, 0 to 207
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Sr No               208 non-null    int64 
 1   DATE                208 non-null    object
 2   Sales               208 non-null    int32 
 3   TV_Spends           208 non-null    int32 
 4   OOH_Spends          208 non-null    int64 
 5   Print_Spends        208 non-null    int32 
 6   FB_Impressions      208 non-null    int32 
 7   Paid_Search_Clicks  208 non-null    int32 
 8   Search_Spends       208 non-null    int64 
 9   competitor_sales_B  208 non-null    int64 
 10  FB_Spends           208 non-null    int32 
 11  events              208 non-null    object
dtypes: int32(6), int64(4), object(2)
memory usage: 14.8+ KB


In [22]:
df2_copy.head()

Sr No        DATE    Sales  TV_Spends  OOH_Spends  Print_Spends  \
0      1  2015-11-23  2754371     167687           0         95463   
1      2  2015-11-30  2584276     214600           0             0   
2      3  2015-12-07  2547386          0      248022          3404   
3      4  2015-12-14  2875220     625877           0        132600   
4      5  2015-12-21  2215953          0      520005             0   

   FB_Impressions  Paid_Search_Clicks  Search_Spends  competitor_sales_B  \
0        72903852                   0              0             8125009   
1        16581099               29511          31000             7901549   
2        49954773               36132          28400             8300197   
3        31649297               36804          31900             8122883   
4         8802269               28401          27100             7105985   

   FB_Spends  events  
0     228213      na  
1      34258      na  
2     127691  event0  
3      84014      na  
4      20687      na

In [23]:
# Change DATE column data type object type to datetime type
df2_copy['DATE'] = df2_copy['DATE'].astype('datetime64[ns]')

In [24]:
df2_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 208 entries, 0 to 207
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Sr No               208 non-null    int64         
 1   DATE                208 non-null    datetime64[ns]
 2   Sales               208 non-null    int32         
 3   TV_Spends           208 non-null    int32         
 4   OOH_Spends          208 non-null    int64         
 5   Print_Spends        208 non-null    int32         
 6   FB_Impressions      208 non-null    int32         
 7   Paid_Search_Clicks  208 non-null    int32         
 8   Search_Spends       208 non-null    int64         
 9   competitor_sales_B  208 non-null    int64         
 10  FB_Spends           208 non-null    int32         
 11  events              208 non-null    object        
dtypes: datetime64[ns](1), int32(6), int64(4), object(1)
memory usage: 14.8+ KB


In [26]:
# change columns positions
titles = list(df2_copy.columns)
titles

['Sr No',
 'DATE',
 'Sales',
 'TV_Spends',
 'OOH_Spends',
 'Print_Spends',
 'FB_Impressions',
 'Paid_Search_Clicks',
 'Search_Spends',
 'competitor_sales_B',
 'FB_Spends',
 'events']

In [27]:
titles[9], titles[10] = titles[10], titles[9]
titles[7], titles[9] = titles[9], titles[7]
titles[6], titles[7] = titles[7], titles[6]
titles

['Sr No',
 'DATE',
 'Sales',
 'TV_Spends',
 'OOH_Spends',
 'Print_Spends',
 'FB_Spends',
 'FB_Impressions',
 'Search_Spends',
 'Paid_Search_Clicks',
 'competitor_sales_B',
 'events']

In [29]:
df2_copy = df2_copy[titles]
df2_copy

Sr No       DATE    Sales  TV_Spends  OOH_Spends  Print_Spends  \
0        1 2015-11-23  2754371     167687           0         95463   
1        2 2015-11-30  2584276     214600           0             0   
2        3 2015-12-07  2547386          0      248022          3404   
3        4 2015-12-14  2875220     625877           0        132600   
4        5 2015-12-21  2215953          0      520005             0   
..     ...        ...      ...        ...         ...           ...   
203    204 2019-10-14  2456240          0       60433        153723   
204    205 2019-10-21  2182825     154917           0             0   
205    206 2019-10-28  2377706      21982       14094         17476   
206    207 2019-11-04  2732825      22453           0         24051   
207    208 2019-11-11  2767788          0           0             0   

     FB_Spends  FB_Impressions  Search_Spends  Paid_Search_Clicks  \
0       228213        72903852              0                   0   
1        34258        16581099          31000               29511   
2       127691        49954773          28400               36132   
3        84014        31649297          31900               36804   
4        20687         8802269          27100               28401   
..         ...             ...            ...                 ...   
203          0               0         112100              152840   
204     133624        56881962         103700              103680   
205          0               0         114700              138387   
206          0               0         134100              151707   
207     366190       127981119              0                   0   

     competitor_sales_B  events  
0               8125009      na  
1               7901549      na  
2               8300197  event0  
3               8122883      na  
4               7105985      na  
..                  ...     ...  
203             7612762      na  
204             6701667      na  
205             7688920      na  
206             8815710      na  
207             9067759      na  

[208 rows x 12 columns]

- In the above section, I replace the data type in the columns Sales, TV_Spends, Print_Spends, FB_Impressions, Paid_Search_Clicks, FB_Spends float to int and DATE column replace into datetime type because the values in the column is high it will showing many numbers after dot our dataset.

In [30]:
df2_copy.to_csv("Assignment_Dataset_Brandintelle_Final_Data.csv",index=False)